### PR TITLE
Enable Bitcoin feature by default on development and nightly channels

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/34731

S&P review: https://github.com/brave/reviews/issues/1438

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/common/features.cc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #include "brave/components/brave_wallet/common/features.h"
 
 #include "base/feature_list.h"
-#include "build/build_config.h"
```
Comment (yrliou) [inline components/brave_wallet/common/features.cc:9]: I thought this was for IS_ANDROID?

Comment (supermassive) [reply components/brave_wallet/common/features.cc:9]: returned back

**Review by yrliou — APPROVED** [review]

**Comment by yrliou** [comment]
Please add a link to the security and privacy issue in this PR or in the issue just for reference, thanks.

File: components/brave_wallet/common/features.cc [hunk 2]
```diff
@@ -26,7 +27,12 @@ BASE_FEATURE(kBraveWalletNftPinningFeature,
 
 BASE_FEATURE(kBraveWalletBitcoinFeature,
              "BraveWalletBitcoin",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+#if BUILDFLAG(ENABLE_BITCOIN_BY_DEFAULT)
```
Comment (bridiver) [inline components/brave_wallet/common/features.cc:30]: features should be enabled/disabled by channel using griffin, not using buildflags

Comment (bridiver) [reply components/brave_wallet/common/features.cc:30]: seems this isn't even used anymore and should be deleted

Comment (supermassive) [reply components/brave_wallet/common/features.cc:30]: > seems this isn't even used anymore and should be deleted

ios team will delete this soon, they're almost with btc support afaik.

Comment (supermassive) [reply components/brave_wallet/common/features.cc:30]: > features should be enabled/disabled by channel using griffin, not using buildflags

- There is no way to enable something in seeds file for dev builds.
- We needed this in nightly and dev builds. Just enabling by default on desktop could get into beta and we didn't want any additional actions to disable it then.
- This approach was already used before https://github.com/brave/brave-core/pull/5898

**Pull request merged**